src/SharpLearning.Optimization/BayesianOptimizer.cs

```diff
@@ -234,68 +262,69 @@ public OptimizerResult[] Optimize(Func<double[], OptimizerResult> functionToMini
                             bestParameterSet = set;
```

Comment:
@jameschch The check `if (score < bestParameterSetScore)` is not thread safe, so we need a lock statement here around the check.

Reply:
Have now created a locked method call for this.